### PR TITLE
fix sdl bugs for uninitialized variables and returns

### PR DESCRIPTION
**Pull request by orilevari**
**Description**: SDL bug fixes for returning and using uninitialized data (warnings C6101 and C6001)

### CODE REVIEW

**Review by martinb35 — APPROVED** [review]

File: winml/adapter/winml_adapter_apis.h [hunk 1]
```diff
@@ -16,7 +16,7 @@ ORT_API(void, ReleaseExecutionProvider, OrtExecutionProvider*);
 ORT_API_STATUS(OverrideSchema);
 
 // OrtEnv methods
-ORT_API_STATUS(EnvConfigureCustomLoggerAndProfiler, _In_ OrtEnv* env, OrtLoggingFunction logging_function, OrtProfilingFunction profiling_function, _In_opt_ void* logger_param, OrtLoggingLevel default_warning_level, _In_ const char* logid, _Outptr_ OrtEnv** out);
+ORT_API_STATUS(EnvConfigureCustomLoggerAndProfiler, _In_ OrtEnv* env, OrtLoggingFunction logging_function, OrtProfilingFunction profiling_function, _In_opt_ void* logger_param, OrtLoggingLevel default_warning_level, _In_ const char* logid);
```
Comment (snnn) [inline winml/adapter/winml_adapter_apis.h:19]: Is it an ABI breaking change?

Comment (orilevari) [reply winml/adapter/winml_adapter_apis.h:19]: Good point, I'll change it back. I originally removed the param because it wasn't used at all.

**Review by martinb35 — APPROVED** [review]

**Pull request merged**